samples/PactV2/EventApi/Provider.Tests/EventAPITests.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using PactNet.AspNetCore.ProviderState;
+using PactNet.Infrastructure.Outputters;
+using PactNet.Verifier;
+using PactNet.Verifier.ProviderState;
+using Provider.Api.Web.Models;
+using Provider.Api.Web.Tests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Provider.Tests
+{
+    public class EventApiTests : IClassFixture<EventApiFixture>
+    {
+        private readonly EventApiFixture fixture;
+        private readonly ITestOutputHelper output;
+        private readonly ProviderStateOptions options;
+        private readonly InMemoryEventRepository repository;
+
+        public EventApiTests(EventApiFixture fixture, ITestOutputHelper output)
+        {
+            this.fixture = fixture;
+            this.output = output;
+            this.options = fixture.GetOptions();
+            this.repository = fixture.GetFakeEventRepository() as InMemoryEventRepository;
+        }
+
+        [Fact]
+        public void EnsureEventApiHonoursPactWithConsumer()
+        {
+            var config = new PactVerifierConfig
+            {
+                Outputters = new List<IOutput>
+                {
+                    new XUnitOutput(this.output)
+                }
+            };
+
+            string pactPath = Path.Combine("..",
+                                           "..",
+                                           "..",
+                                           "..",
+                                           "Consumer.Tests",
+                                           "pacts",
+                                           "Event API Consumer-Event API.json");
+
+            //Act / Assert
+            IPactVerifier pactVerifier = new PactVerifier(config);
+            pactVerifier
+                .ServiceProvider("Event API", this.fixture.ServerUri)
+                .HonoursPactWith("Event API Consumer")
+                .FromPactFile(new FileInfo(pactPath))
+                .WithProviderStateUrl(new Uri(this.fixture.ServerUri, options.RouteProviderState))
+                .WithProviderStates(providerStates =>
+                {
+                    providerStates
+                        .Add(
+                            "there are events with ids '45D80D13-D5A2-48D7-8353-CBB4C0EAABF5', '83F9262F-28F1-4703-AB1A-8CFD9E8249C9' and '3E83A96B-2A0C-49B1-9959-26DF23F83AEB'",
+                            this.InsertEventsIntoDatabase, StateAction.Setup)
```

Comment:
I'm not a big fan of the enum on the end here, especially because it's optional (and defaults to `Setup`?). I'd prefer the methods themselves to indicate what they were doing, like:

```csharp
states.AddSetup("my provider state", () => db.InsertData())
      .AddTeardown("my provider state", () => db.ClearData());
```

Reply:
Good point